### PR TITLE
Look up path to client certificate relative to home directory

### PR DESCRIPTION
**Pull request by greg-1-anderson**
See #88.

### CODE REVIEW

File: modules/pantheon/pantheon_apachesolr/pantheon_apachesolr.module [hunk 1]
```diff
@@ -157,6 +157,17 @@ function pantheon_apachesolr_curlopts() {
   );
 }
 
+function pantheon_apachesolr_certs_dir() {
+  if (isset($_ENV['PANTHEON_ENVIRONMENT'])) {
+    return $_SERVER['HOME'] . '/certs';
+  }
+  return '../certs';
```
Comment (greg-1-anderson) [inline modules/pantheon/pantheon_apachesolr/pantheon_apachesolr.module]: It seems unlikely that we would want to support a local development environment for Pantheon ApacheSolr; if we did, though, we could look at `git rev-parse --show-toplevel` here to find the root of the git repository, and perhaps look for a `certs` dir one level above that.

It is probably sufficient to support only the Pantheon environment here, though.

Comment (joshkoenig) [inline modules/pantheon/pantheon_apachesolr/pantheon_apachesolr.module]: Pantheon only please. This whole module should deactivate itself if `$_ENV['PANTHEON_ENVIRONMENT']` is undefined.

**Comment by greg-1-anderson** [comment]
With https://github.com/pantheon-cookbooks/endpoint/pull/280 merged, it is now possible to use either `$_SERVER['HOME']` or `$_ENV['HOME']` to find the home directory here.  However, I think that we should stick with `$_SERVER['HOME']`.  The reason is that Drush sets `$_SERVER['HOME']`, so this value works fine from a cli tool, whereas `$_ENV['HOME']` typically is not set, e.g. on other hosting providers, from Drush on your local system, etc. 

For an explanation of why this is, see [variables_order](http://php.net/manual/en/ini.core.php#ini.variables-order). The documentation claims the default value is "EGPCS", but if you look in php.ini, you see:
```
; variables_order
;   Default Value: "EGPCS"
;   Development Value: "GPCS"
;   Production Value: "GPCS"
```
Ergo, in most scenarios you get "GPCS", which means no "E", which means that $_ENV is not set in most default environments.  In contrast, `$_SERVER["HOME"]` is pretty much universally available.

Having HOME in .env on Pantheon is still a good idea to ensure that the HOME environment variable is set for other contexts (non nginx, non-drush).

**Comment by joshkoenig** [comment]
I'm not sure why we are concerned with how this code runs on other providers. It only works on Pantheon.

I prefer we use `$_ENV['HOME']` any time we want to insure consistent values in CLI or Web execution. We take responsibility for this, and can guarantee consistency. 

Otherwise we depending on individual CLI tools like Drush to handle populating `$_SERVER` for us. It's nice that Drush does this, and that Greg is on the team, but I don't know if this is a safe assumption for other contexts, and there's value in having a _standard_ way that we reference variables.

**Comment by greg-1-anderson** [comment]
I'm concerned with other providers because it seems that overall, `$_SERVER["HOME"]` is more standard than `$_ENV["HOME"]`; the later rarely works.

But, if you prefer a _Pantheon way_, then switching this to `$_ENV["HOME"]` is fine, and works.

**Comment by greg-1-anderson** [comment]
Adjusted to use `$_ENV["HOME"]`

**Comment by joshkoenig** [comment]
Thanks. Again, this code won't ever work on another provider. It's our code for our platform only, so sticking to our standard seems to be the right thing to do.

**Comment by greg-1-anderson** [comment]
Fixed in #92

**Comment by greg-1-anderson** [comment]
Deployed.

**Pull request closed without merging**